src/main/scala/viper/silver/plugin/standard/refute/RefutePlugin.scala

```diff
@@ -0,0 +1,78 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2022 ETH Zurich.
+
+package viper.silver.plugin.standard.refute
+
+import fastparse.P
+import viper.silver.ast.utility.ViperStrategy
+import viper.silver.ast._
+import viper.silver.parser.FastParser.{FP, exp, keyword, whitespace}
+import viper.silver.parser.ParserExtension
+import viper.silver.plugin.{ParserPluginTemplate, SilverPlugin}
+import viper.silver.verifier._
+import viper.silver.verifier.errors.AssertFailed
+
+class RefutePlugin extends SilverPlugin with ParserPluginTemplate {
+
+  /** Keyword used to define refute statements. */
+  private val RefuteKeyword: String = "refute"
+
+  private var RefuteAsserts: Map[Position, Refute] = Map()
+
+  /** Parser for refute statements. */
+  def refute[_: P]: P[PRefute] =
+    FP(keyword(RefuteKeyword) ~/ exp).map{ case (pos, e) => PRefute(e)(pos) }
+
+  /** Add refute to the parser. */
+  override def beforeParse(input: String, isImported: Boolean): String = {
+    // Add new keyword
+    ParserExtension.addNewKeywords(Set[String](RefuteKeyword))
+    // Add new parser to the precondition
+    ParserExtension.addNewStmtAtEnd(refute(_))
+    input
+  }
+
+  /**
+   * Remove refute statements from the AST and add them as non-det asserts.
+   * The ⭐ is nice since such a variable name cannot be parsed, but will it cause issues?
```

Comment:
Apart from looking bad (different, coloured font), this might cause some Unicode problems in terminals? I think Windows cmd is especially bad with this…

Reply:
Windows' latest terminal can handle Unicode, but I'd nonetheless refrain, to reduce the risk of problems (e.g. in some tool further down such as Boogie, Z3, CVC5, the Axiom Profiler, ...).

I also find it a bit mean to use a representation that users might be exposed to, but that they then cannot use.

Reply:
Users can pretty-print the program the backend ultimately verifiers. My concern is that they'll get a program that they then cannot verify because it doesn't parse. Can that happen?

Reply:
I think for plugins we could choose some kind of naming convention for temporaries, e.g. `__plugin_refute_<counter>` (and `__plugin_X_Y` for other plugins). This way it is clear what is going on from reading the desugared AST. Of course a user would be able to declare a conflicting identifier but given the prefix it would be clearly their intention.

Reply:
Makes sense, I'll use Aurel's suggestion - now the pretty-printed program won't be parsable only if the user declares a conflicting identifier (which seems ok to me). If that is something we really care about, we could add a github CI which takes the `printTranslatedProgram` output and reruns it again checking that the result is identical, not sure if it's worth the extra CI time though.

Reply:
Sounds good, thanks. That's also what other plugins do.